docs: Add pandas strictness API difference (#21312)
pola-rs/polars: docs/source/user-guide/migration/pandas.md

### Polars adheres to the Apache Arrow memory format to represent data in memory while pandas uses NumPy arrays

Polars represents data in memory according to the Arrow memory spec while pandas by default
represents data in memory with NumPy arrays. Apache Arrow is an emerging standard for in-memory
columnar analytics that can accelerate data load times, reduce memory usage and accelerate
calculations.

Polars can convert data to NumPy format with the `to_numpy` method.

### Polars has more support for parallel operations than pandas

Polars exploits the strong support for concurrency in Rust to run many operations in parallel. While
some operations in pandas are multi-threaded the core of the library is single-threaded and an
additional library such as `Dask` must be used to parallelize operations. Polars is faster than all

> open source solutions that paralellize pandas code.

Reviewer: ```suggestion
open source solutions that parallelize pandas code.
```

`Dask` also supports lazy evaluation when it generates a query plan.

### Polars is strict

> Polars is strict about data types. Data type resolution in Polars are dependent on the operation

Reviewer: ```suggestion
Polars is strict about data types. Data type resolution in Polars is dependent on the operation
```

graph, whereas pandas converts types loosely (e.g. new missing data can lead to integer columns

> being converted to floats). This strictness leads to less bugs and more predicatable behavior.

Reviewer: ```suggestion
being converted to floats). This strictness leads to fewer bugs and more predictable behavior.
```

### Polars has a more verstatile API

Polars is built on expressions and allows expression inputs in almost all operations. This means
that when you understand how expressions work, your knowledge in Polars extrapolates. Pandas doesn't

> have an expression system and often requires Python lambda's to express the complexity you want.

Reviewer: ```suggestion
have an expression system and often requires Python `lambda`s to express the complexity you want.
```

> Polars sees the requirement of a Python lambda as a lack of expressiveness of its API, and tries to

Reviewer: ```suggestion
Polars sees the requirement of a Python `lambda` as a lack of expressiveness of its API, and tries to
```
